ui-manchette/src/components/Manchette.tsx

```diff
@@ -1,51 +1,217 @@
-import React from 'react';
+import React, { useEffect, useRef } from 'react';
 import OperationalPointList from './OperationalPointList';
-import { OperationalPointType } from '../types';
+import { OperationalPointType, ProjectPathTrainResult, StyledOperationalPointType } from '../types';
 import { ZoomIn, ZoomOut } from '@osrd-project/ui-icons';
-
+import { SpaceTimeChart, PathLayer } from '@osrd-project/ui-spacetimechart';
+import { calcOperationalPointsToDisplay, operationalPointsHeight } from './helpers';
+import {
+  OperationalPoint,
+  PathData,
+  SpaceScale,
+} from '@osrd-project/ui-spacetimechart/dist/lib/types';
+import {
+  BASE_KM_HEIGHT,
+  BASE_OP_HEIGHT,
+  INITIAL_OP_LIST_HEIGHT,
+  INITIAL_SPACE_TIME_CHART_HEIGHT,
+} from './consts';
+import { getDiff } from '../utils/vector';
+import { useIsOverflow } from '../hooks/isOverFlowed';
 type ManchetteProps = {
   operationalPoints: OperationalPointType[];
-  children?: React.ReactNode;
+  projectPathTrainResult: ProjectPathTrainResult[];
   isProportionnal?: boolean;
 };
 
 const Manchette: React.FC<ManchetteProps> = ({
   operationalPoints,
-  children,
+  projectPathTrainResult,
   isProportionnal = false,
 }) => {
-  const [zoom, setZoom] = React.useState(1);
+  const manchette = useRef<HTMLDivElement>(null);
+
+  const [zoomY, setZoomY] = React.useState(1);
+
+  const xZoomLevel = 1;
+
+  const [xOffset, setXOffset] = React.useState(0);
+
+  const [yOffset, setYOffset] = React.useState(0);
+  const [panning, setPanning] = React.useState<{ initialOffset: { x: number; y: number } } | null>(
+    null
+  );
+
+  const [scrollPosition, setScrollPosition] = React.useState(0);
+
+  const [operationalPointsToDisplay, setOperationalPointsToDisplay] = React.useState<
+    StyledOperationalPointType[]
+  >([]);
+
+  const [operationalPointsChart, setOperationalPointsChart] = React.useState<OperationalPoint[]>(
+    []
+  );
+
+  const [panY, setPanY] = React.useState<boolean>(false);
+
+  const [scales, setScales] = React.useState<SpaceScale[]>([]);
+
+  const zoomYIn = () => {
+    if (zoomY < 10) setZoomY(zoomY + 0.5);
+  };
 
-  const zoomIn = () => {
-    if (zoom < 10) setZoom(zoom + 0.5);
+  const zoomYOut = () => {
+    if (zoomY > 1) setZoomY(zoomY - 0.5);
   };
 
-  const zoomOut = () => {
-    if (zoom > 1) setZoom(zoom - 0.5);
+  useIsOverflow(manchette, (isOverflowFromCallback) => {
+    setPanY(isOverflowFromCallback);
+  });
+
+  const paths: (PathData & { color: string })[] = projectPathTrainResult.map((path, index) => ({
+    id: path.departure_time,
+    label: `Train ${index + 1}`,
+    color: '#000000',
+    points: path.space_time_curves.flatMap((curve) => {
+      const dataPoints = [];
+      for (let i = 0; i < curve.positions.length; i++) {
+        dataPoints.push({
+          time: curve.times[i] + +new Date(path.departure_time),
+          position: curve.positions[i],
+        });
+      }
+      return dataPoints;
+    }),
+  }));
```

Comment:
I suggest here that to refactore the points mapping to directly return the result and wrapped it in `useMemo` for performance optimization, improving code readability and conciseness.
```suggestion
const paths = useMemo(() => projectPathTrainResult.map((path, index) => ({
    id: path.departure_time,
    label: `Train ${index + 1}`,
    color: '#000000',
    points: path.space_time_curves.flatMap((curve) => {
      return curve.positions.map((position, i) => ({
        time: curve.times[i] + +new Date(path.departure_time),
        position,
      }));
    }),
  })), [projectPathTrainResult]);
```

Reply:
done i created a new hook